src/Utilities/FileSystem.cpp

```diff
@@ -78,35 +78,33 @@ std::string get_absolute_path(const std::string& rel_path) {
           << rel_path << "'.");
       // LCOV_EXCL_STOP
     }
-    ERROR("Failed to get absolute path for an unknown reason.");
+    ERROR("Failed to get absolute path for an unknown reason: " << errno);
   }
   std::string return_name(name);
-  delete[] name;
+  free(name);
   return return_name;
 }
 
-bool create_directory(std::string dir, const double wait_time,
-                      const size_t num_tries, const bool terminate_on_failure) {
+void create_directory(const std::string& in_dir, const double wait_time,
+                      const size_t num_tries) {
   // each time we fail to create a directory, we increase the wait_time by this
   // factor
   static constexpr double wait_time_increase_factor = 1.1;
-  if (dir.empty()) {
+  if (in_dir.empty()) {
     ERROR("Cannot create a directory that has no name");
   }
-  // Remove the trailing slashes on the directory name
-  while (not dir.empty() and dir[dir.size() - 1] == '/') {
-    dir.erase(dir.size() - 1);
-  }
-  if (dir.empty()) {
-    ERROR("Cannot create the root directory '/' (possibly multiple slashes)");
+
+  std::string dir = filesystem::dirname(in_dir);
+  if ("/"s == dir) {
+    return;
```

Comment:
While I think it is unlikely that we will want to create any directories directly in /, I don't see why we are just returning early if someone tries.